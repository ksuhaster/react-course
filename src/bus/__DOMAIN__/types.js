export const types = Object.freeze({
    DOMAIN_ACTION: 'DOMAIN_ACTION'
});