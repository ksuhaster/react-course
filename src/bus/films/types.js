export const types = Object.freeze({

    // === PEOPLE ===

    // Sync
    FILMS_START_FETCHING: 'FILMS_START_FETCHING',
    FILMS_STOP_FETCHING: 'FILMS_STOP_FETCHING',
    FILMS_FILL: 'FILMS_FILL',
    FILMS_SET_FETCHING_ERROR: 'FILMS_SET_FETCHING_ERROR',

    // Async
    FILMS_FETCH_ASYNC: 'FILMS_FETCH_ASYNC',

    // === PERSON ===

    // Sync
    FILM_START_FETCHING: 'FILM_START_FETCHING',
    FILM_STOP_FETCHING: 'FILM_STOP_FETCHING',
    FILM_FILL: 'FILM_FILL',
    FILM_SET_FETCHING_ERROR: 'FILM_SET_FETCHING_ERROR',

    // Async
    FILM_FETCH_ASYNC: 'FILM_FETCH_ASYNC',
});
