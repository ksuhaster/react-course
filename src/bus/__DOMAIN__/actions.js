// Types
import { types } from './types';

export const domainActions = Object.freeze({
    domainAction: () => {
        return {
            type: types.DOMAIN_ACTION
        }
    }
});