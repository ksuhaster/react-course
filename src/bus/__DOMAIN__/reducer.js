// Types
import { types } from './types';

const initialState = {
    property: ''
};

export const domainReducer = ( state = initialState, { type } ) => {
    switch (type) {
        case types.DOMAIN_ACTION:
            return { ...state, property: '' };
    
        default:
            return state;
    }
}