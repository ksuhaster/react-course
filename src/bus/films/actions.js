// Types
import { types } from './types';

import { api } from '../../api';

export const filmsActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.FILMS_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.FILMS_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.FILMS_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FILMS_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    fetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.FILMS_FETCH_ASYNC
        });

        dispatch(filmsActions.startFetching());

        const response = await api.films.fetch();

        if (response.status === 200) {
            const { results } = await response.json();
            dispatch(filmsActions.fill(results));
        } else {
            const error = {
                status: response.status
            };

            dispatch(filmsActions.setFetchingError(error));
        }

        dispatch(filmsActions.stopFetching());
    }
});


export const filmActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.FILM_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.FILM_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.FILM_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FILM_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    fetchAsync: (id) => async (dispatch) => {
        dispatch({
            type: types.FILM_FETCH_ASYNC
        });

        dispatch(filmActions.startFetching());

        const response = await api.film.fetch(id);

        if (response.status === 200) {
            const results = await response.json();
            dispatch(filmActions.fill(results));
        } else {
            const error = {
                status: response.status
            };

            dispatch(filmActions.setFetchingError(error));
        }

        dispatch(filmActions.stopFetching());
    }
});