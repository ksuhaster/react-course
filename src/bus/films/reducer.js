// Types
import { types } from './types';

const initialState = {
    data: [],
    isFetching: false,
    error: false
};

export const filmsReducer = ( state = initialState, { type, payload } ) => {
    switch (type) {
        case types.FILMS_START_FETCHING:
            return { ...state, isFetching: true };
        case types.FILMS_STOP_FETCHING:
            return { ...state, isFetching: false };
        case types.FILMS_SET_FETCHING_ERROR:
            return { ...state, error: payload };
        case types.FILMS_FILL:
            return { ...state, data: payload };

        default:
            return state;
    }
};

export const filmReducer = ( state = initialState, { type, payload } ) => {
    switch (type) {
        case types.FILM_START_FETCHING:
            return { ...state, isFetching: true };
        case types.FILM_STOP_FETCHING:
            return { ...state, isFetching: false };
        case types.FILM_SET_FETCHING_ERROR:
            return { ...state, error: payload };
        case types.FILM_FILL:
            return { ...state, data: payload };

        default:
            return state;
    }
};