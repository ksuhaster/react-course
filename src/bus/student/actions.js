// Types
import { types } from './types';

export const studentActions = Object.freeze({
    setStudent: (payload) => {
        return {
            type: types.SET_NAME,
            payload
        }
    }
});