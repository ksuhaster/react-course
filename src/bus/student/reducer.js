// Types
import { types } from './types';

const initialState = {
    name: ''
};

export const studentReducer = ( state = initialState, { type, payload } ) => {
    switch (type) {
        case types.SET_NAME:
            return { ...state, name: payload };
    
        default:
            return state;
    }
}