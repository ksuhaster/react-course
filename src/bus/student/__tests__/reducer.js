import { studentReducer } from '../reducer';
import { studentActions } from '../actions';
import { types } from "../types";


describe('student reducer:', () => {
    test('should return initial state by default', () => {
        expect(studentReducer(void 0, {})).toMatchSnapshot();
    });

    test(`should handle ${types.SET_STUDENT} action`, () => {
        expect(studentReducer(void 0, studentActions.setStudent())).toMatchSnapshot();
    });
});

