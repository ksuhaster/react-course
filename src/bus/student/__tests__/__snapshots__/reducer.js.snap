// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`student reducer: should handle SET_STUDENT action 1`] = `
Object {
  "age": "",
  "confirmpassword": "",
  "email": "",
  "firstName": "",
  "formIsNew": true,
  "password": "",
  "sex": "",
  "speciality": "",
  "surname": "",
}
`;

exports[`student reducer: should return initial state by default 1`] = `
Object {
  "age": "",
  "confirmpassword": "",
  "email": "",
  "firstName": "",
  "formIsNew": true,
  "password": "",
  "sex": "",
  "speciality": "",
  "surname": "",
}
`;
